src/crunchy_copy.py

```diff
@@ -130,188 +141,220 @@ def delete_all_files_in_dir(source_dir):
     return h, m, s
 
 
-def summarize(start, finish, download_finishes, upload_finishes):
+def summarize(start, finish):
     print("SUMMARY:")
     print(f"Start: {start}")
     print(f"Finish: {finish}")
     total_h, total_m, total_s = seconds_to_readable((finish - start).total_seconds())
     print(f"Total Duration: {total_h} hrs, {total_m} minutes, {total_s} seconds")
 
 
-def signal_dead_mans_snitch():
+def signal_dead_mans_snitch(cluster: str):
     print("Signaling Dead Man's Snitch...")
-    if args.backend not in STAGING_BACKENDS:
+    if cluster not in STAGING_BACKENDS:
         with open("./src/backend-snitch-map.json") as json_map:
             backend_snitch_map = json.load(json_map)
-        res = requests.post(backend_snitch_map[args.backend], data={"m": "Completed"})
+        res = requests.post(backend_snitch_map[cluster], data={"m": "Completed"})
         return res
     else:
         return
 
 
-def main():
-    tz = ZoneInfo("US/Eastern")
-    script_start = datetime.utcnow().replace(tzinfo=tz)
-    # Establish connection to AspirEDU's S3 Resource
-    aspire_s3_resource, aspire_s3_client = get_s3(
-        ASPIRE_AWS_ACCESS_KEY_ID, ASPIRE_AWS_SECRET_ACCESS_KEY
-    )
-
-    # Connect to AspirEDU backup Bucket
-    aspire_bucket = (
-        aspire_s3_resource.Bucket("aspiredu-pgbackups")
-        if args.backend not in AU_BACKENDS
-        else aspire_s3_resource.Bucket("aspiredu-pgbackups-au")
-    )
-
-    # Get Cluster information from CrunchyBridge
-    clusters = get_crunchy_clusters()
-    for cluster in clusters:
-        if cluster["name"] != args.backend:
-            continue
-        download_path = f"{LOCAL_TEMP_DOWNLOADS_PATH}{cluster['name']}"
-        backup_info = get_cluster_backup_info(cluster["id"])
+class CrunchyCopy:
+    def __init__(
+        self, bucket_name: str, cluster_name: str, backup_target: str, dry_run: bool = False
+    ):
+        """
+
+        :param bucket_name: The name of the bucket to copy to.
+        :param cluster_name: The name of the CrunchyBridge cluster that the backup is from.
+        :param backup_target: The date prefix for the backup we're targeting, such as `20200101`
+        """
+        self.s3_resource, self.s3 = get_s3(ASPIRE_AWS_ACCESS_KEY_ID, ASPIRE_AWS_SECRET_ACCESS_KEY)
+        self.bucket = self.s3_resource.Bucket(bucket_name)
+        self.backup_target = backup_target
+        self.cluster = self.get_cluster(cluster_name)
+        self.backup_info = get_cluster_backup_info(
+            self.cluster["id"], backup_target=self.backup_target
+        )
+        self.dry_run = dry_run
+
+    @classmethod
+    def get_cluster(cls, cluster_name: str) -> dict:
+        """Find the cluster from the CrunchyBridge API with the given name"""
+        for cluster in get_crunchy_clusters():
+            if cluster["name"] == cluster_name:
+                return cluster
+        raise CantFindCrunchyBridgeCluster("Could not find cluster with the given name")
```

Comment:
Could be helpful to add the cluster name that couldn't be found in the error string here.

Reply:
That should be in the locals of Sentry. This will batch the errors together by not having that specification.

Reply:
Got it, thanks for the explanation!